typings/index.d.ts

```diff
@@ -1939,13 +1949,22 @@ declare module 'discord.js' {
   export class GuildMemberManager extends BaseManager<Snowflake, GuildMember, GuildMemberResolvable> {
     constructor(guild: Guild, iterable?: Iterable<any>);
     public guild: Guild;
-    public ban(user: UserResolvable, options?: BanOptions): Promise<GuildMember | User | Snowflake>;
     public fetch(
       options: UserResolvable | FetchMemberOptions | (FetchMembersOptions & { user: UserResolvable }),
     ): Promise<GuildMember>;
     public fetch(options?: FetchMembersOptions): Promise<Collection<Snowflake, GuildMember>>;
     public prune(options: GuildPruneMembersOptions & { dry?: false; count: false }): Promise<null>;
     public prune(options?: GuildPruneMembersOptions): Promise<number>;
+  }
+
+  export class GuildBanManager extends BaseManager<Snowflake, GuildBan, GuildBanResolvable> {
+    constructor(guild: Guild, iterable?: Iterable<any>);
+    public guild: Guild;
+    public ban(user: UserResolvable, options?: BanOptions): Promise<GuildMember | User | Snowflake>;
+    public fetch(
+      options: UserResolvable | FetchBanOptions | (UserResolvable & { user: UserResolvable }),
+    ): Promise<GuildBan>;
+    public fetch(): Promise<Collection<Snowflake, GuildBan>>;
```

Comment:
```suggestion
    public fetch(
      options: UserResolvable | FetchBanOptions | (UserResolvable & { user: UserResolvable }),
    ): Promise<GuildBan>;
    public fetch(
      options: FetchBanOptions & { cache: boolean },
    ): Promise<GuildBan | Collection<Snowflake, GuildBan>>;
    public fetch(): Promise<Collection<Snowflake, GuildBan>>;
```
if `cache` is set, `fetch` may return `_fetchMany`.
```js
    if (options.user) {
      options.user = this.client.users.resolveID(options.user);
    }
    if (!options.user) {
      if ('cache' in options) return this._fetchMany(options.cache);
      return Promise.reject(new Error('FETCH_BAN_RESOLVE_ID'));
    }
    return this._fetchSingle(options);
```
Also, I don't get the role of `(UserResolvable & { user: UserResolvable })`. It's don't seems to be used.

Reply:
> if cache is set, fetch may return _fetchMany.

well yes, that is intended, if user can't be resolved but `cache` is present _fetchMany is returned
```js
g.bans.fetch({ cache: true }); //this fetches all bans and caches them
g.bans.fetch({ cache: false }); //this fetches all bans without caching
g.bans.fetch("351871113346809860"); //this fetches a ban and caches it
g.bans.fetch({ user: "351871113346809860", cache: false }); //this fetches a ban but doesn't cache it
```
if this code didn't exist
```js
if ('cache' in options) return this._fetchMany(options.cache);
```
first 2 examples would error with `FETCH_BAN_RESOLVE_ID`

Reply:
I've reworked fetch options in 1f7a901 so I'll mark this as resolved